contracts/utils/EmergencyBrake.sol

```diff
@@ -82,41 +81,41 @@ contract EmergencyBrake is AccessControl, IEmergencyBrake {
     {   
-        require(plans[target].state == State.PLANNED, "Target not planned for");
-        require(toAdd.signature != ROOT, "Can't remove ROOT");
-        Permission[] memory _permissions = plans[target].permissions;
-        bytes32 toIndex = _persmissionToId(toAdd.contact, toAdd.signature);
-        require(plans[target].index[toIndex] == 0, "Permission set already in plan");
-        uint256 planId = _permissions.length;
-        plans[target].permissions.push(toAdd);
-        plans[target].index[toIndex] = planId;
+        Plan storage _plan = plans[target];
+        require(_plan.state == State.PLANNED, "Target not planned for");
+        require(newPermission.signature != ROOT, "Can't remove ROOT");
+        Permission[] memory _permissions = _plan.permissions;
+        bytes32 newId = _permissionToId(newPermission);
+        require(_plan.index[newId] == 0, "Permission set already in plan");
+        _plan.permissions.push(newPermission);
+        _plan.index[newId] = _permissions.length - 1;
         
-        emit AddedTo(target, toAdd);
+        emit PermissionAdded(target, newPermission);
     }
 
     /// @dev remove a permission set from an existing plan
     /// @dev retains the order of permissions and updates their index
     /// @param target address wuth auth privileges on a contract and a plan exists for
-    function removeFromPlan(address target, Permission calldata toRemove) 
+    function removeFromPlan(address target, Permission memory permissionOut) 
         external override auth
     {   
-        require(plans[target].state == State.PLANNED, "Target not planned for");
-        Permission[] memory _permissions = plans[target].permissions;
-        bytes32 idToRemove = _persmissionToId(toRemove.contact, toRemove.signature); 
-        uint256 indexToReplace = plans[target].index[idToRemove];
-        require(indexToReplace > 0, "Permission set not planned");
-        --indexToReplace;
-        uint256 replacement = _permissions.length - 1;
-        bytes32 idToReindex = _persmissionToId(_permissions[replacement].contact, _permissions[replacement].signature);
-        plans[target].permissions[indexToReplace] = _permissions[replacement];
-        plans[target].index[idToRemove] = 0;
-        plans[target].index[idToReindex] = indexToReplace;
-        plans[target].permissions.pop();
-        emit RemovedFrom(target, toRemove);
+        Plan storage _plan = plans[target];
+        require(_plan.state == State.PLANNED, "Target not planned for");
+        Permission[] memory _permissions = _plan.permissions;
+        bytes32 idOut = _permissionToId(permissionOut); 
+        uint256 indexOut = plans[target].index[idOut];
+        require(_permissionToId(_plan.permissions[indexOut]) == idOut, "Permission set not planned"); // indexOut might be zero if permissionOut is not in the plan, or if we want to remove the permission at position zero. We just check that the permission we are removing is the one intended.
```

Comment:
This is much better and more clever way than how I'd been doing it before. 

```bytes32 idCheck = _permissionToId(_plan.permissions[indexOut]); 
require(idCheck == idOut, "Permission set not planned");```
This is the only thing I could thing of to make it more obvious but I don't think it's necessary, especially with your commentary.